tensorflow/core/kernels/non_max_suppression_op.cc

```diff
@@ -569,6 +582,8 @@ class NonMaxSuppressionV2Op : public OpKernel {
     int num_boxes = 0;
     ParseAndCheckBoxSizes(context, boxes, &num_boxes);
     CheckScoreSizes(context, num_boxes, scores);
+    // check if the boxes have logical coordinates
+    OP_REQUIRES(context, CheckBoxesCoordinates<T>(boxes, num_boxes), errors::InvalidArgument("boxes coordinates shouldn't have x1=x2 or y1=y2"));
```

Comment:
You should replace all the checks in the same way, not just the one I commented on on the previous review.

Also, the comments above these checks are redundant, please remove.